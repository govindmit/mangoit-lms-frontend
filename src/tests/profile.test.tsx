import React from "react";
import { render, screen } from "@testing-library/react";
import Profile from "@/pages/user/profile";
import "@testing-library/jest-dom";

describe("Profile", () => {
  it("renders the Profile component", () => {
    render(<Profile />);
    const title = screen.getByText("Profile");
    expect(title).toBeInTheDocument();

    const { container } = render(<Profile />);
    // expect(container).toMatchSnapshot();
  });

  it("Should render submit button", async () => {
    render(<Profile />);

    const main = screen.queryByTestId("step-1");
    expect(main).toBeInTheDocument();
    // expect(screen.queryByTestId("step-1")).toBeInTheDocument();
  });
});

//check for submit button
// const button = screen.getByRole("Button", { name: "Update Profile" });
// expect(button).toBeInTheDocument();
// expect(button).not.toBeDisabled();

// expect(screen.queryByTestId("iconbtn")).not.toBeInTheDocument();
// expect(screen.findByText('Update Profile')).toBeInTheDocument()
// expect(screen.getByTestId("iconbtn")).toBeInTheDocument()
// const btn = screen.getByRole("button"); // get the button (pressable)
// fireEvent.click(btn); // click it
