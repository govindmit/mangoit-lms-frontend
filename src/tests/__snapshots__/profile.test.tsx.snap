// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Should render submit button 1`] = `
<div>
  <div
    class="combineContentAndSidebar MuiBox-root css-0"
  >
    <div
      class="siteBodyContainer MuiBox-root css-0"
    >
      <div
        class="MuiStack-root css-q9mojd-MuiStack-root"
      >
        <div
          class="MuiStack-root css-nen11g-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1lwbda4-MuiStack-root"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    color="inherit"
                    href="https://mangoit-lms.mangoitsol.com/user/dashboard"
                    style="color: rgb(125, 134, 165); text-decoration: none;"
                  >
                    Home
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  ›
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    color="inherit"
                    href="#"
                    style="color: rgb(232, 102, 27); text-decoration: none;"
                  >
                    Profile
                  </a>
                </li>
              </ol>
            </nav>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            style="font-weight: bold; color: rgb(51, 51, 51);"
          >
            USER PROFILE
          </h5>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <form
            autocomplete="off"
            class="MuiBox-root css-0"
            method="POST"
            novalidate=""
          >
            Record not found
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</div>
`;

exports[`Profile renders the Profile component 1`] = `
<div>
  <div
    class="combineContentAndSidebar MuiBox-root css-0"
  >
    <div
      class="siteBodyContainer MuiBox-root css-0"
    >
      <div
        class="MuiStack-root css-q9mojd-MuiStack-root"
      >
        <div
          class="MuiStack-root css-nen11g-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1lwbda4-MuiStack-root"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    color="inherit"
                    href="https://mangoit-lms.mangoitsol.com/user/dashboard"
                    style="color: rgb(125, 134, 165); text-decoration: none;"
                  >
                    Home
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  ›
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    color="inherit"
                    href="#"
                    style="color: rgb(232, 102, 27); text-decoration: none;"
                  >
                    Profile
                  </a>
                </li>
              </ol>
            </nav>
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            style="font-weight: bold; color: rgb(51, 51, 51);"
          >
            USER PROFILE
          </h5>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <form
            autocomplete="off"
            class="MuiBox-root css-0"
            method="POST"
            novalidate=""
          >
            Record not found
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
